lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -6,6 +6,32 @@ class JsonApi < Base
         autoload :PaginationLinks
         autoload :FragmentCache
 
+        module ApiObjects
+          # Make JSON API top-level jsonapi member opt-in
+          # ref: http://jsonapi.org/format/#document-top-level
+          ActiveModel::Serializer.config.jsonapi_include_toplevel_member = false
+          ActiveModel::Serializer.config.jsonapi_version = '1.0'
+          ActiveModel::Serializer.config.jsonapi_toplevel_meta = {}
+          module JsonApi
+            module_function
+
+            def include_member?
+              ActiveModel::Serializer.config.jsonapi_include_toplevel_member
+            end
+
+            def add!(document)
```

Comment:
I think `do_some_magic(foo)` is disingenous since the method is `JsonApiObject.add!(foo)`, it's clear you're mutating foo to addd a JsonApiObject.